src/lib/dma.c

```diff
@@ -189,10 +189,13 @@ void dma_sg_free(struct dma_sg_elem_array *elem_array)
 	dma_sg_init(elem_array);
 }
 
-void dma_buffer_copy_from(struct comp_buffer *source, uint32_t source_bytes,
-			  struct comp_buffer *sink, uint32_t sink_bytes,
-			  dma_process_func process, uint32_t samples)
+void dma_buffer_copy_from(struct comp_buffer *source, struct comp_buffer *sink,
+			  dma_process_func process, uint32_t source_bytes)
 {
+	uint32_t samples = source_bytes /
+			   audio_stream_sample_bytes(&source->stream);
```

Comment:
I was good with this patch since it used bytes instead of samples, which is a good thing to do when dealing with DMAs, but this re-introduces samples which may not be correct in all cases if the data is not PCM.

Reply:
I will remove this samples number calculation from this place, but in following PR, let me do it step-by-step. Even this PR needs changes from https://github.com/thesofproject/sof/pull/3174, I wouldn't like to make too long PR dependency chain

Reply:
ok, looks like we are in agreement on directions, fine with me if we do this in steps, as long as each PR doesn't break because the others are missing.